deepchecks/vision/checks/performance/confusion_matrix.py

```diff
@@ -0,0 +1,187 @@
+
+    def initialize_run(self, context: Context):
+        """Initialize run by creating an empty matrix the size of the data."""
+        context.assert_task_type(TaskType.CLASSIFICATION, TaskType.OBJECT_DETECTION)
+        dataset: VisionData = context.train
+
+        self.task_type = dataset.task_type
+        self.num_classes = dataset.get_num_classes()
+
+        matrix_size = self.num_classes if self.task_type == TaskType.CLASSIFICATION else self.num_classes + 1
+
+        self.matrix = np.zeros((matrix_size, matrix_size))
+
+    def update(self, context: Context, batch: Any, dataset_name: str = 'train'):
+        """Add batch to confusion matrix."""
+        if dataset_name == 'train':
+            dataset = context.train
+        else:
+            dataset = context.test
+
+        labels = dataset.label_transformer(batch[1])
+
+        predictions = context.infer(batch[0])
+
+        if self.prediction_formatter:
+            predictions = self.prediction_formatter(predictions)
+
+        if self.task_type == TaskType.CLASSIFICATION:
+            self.update_classification(predictions, labels)
+        elif self.task_type == TaskType.OBJECT_DETECTION:
+            self.update_object_detection(predictions, labels)
+
+    def compute(self, context: Context) -> CheckResult:
+        display_confusion_matrix, categories = filter_confusion_matrix(self.matrix, self.categories_to_display)
+
+        description = ''
+
+        if self.task_type == TaskType.OBJECT_DETECTION:
+            if self.num_classes == categories[-1]:
+                categories[-1] = 'not found'
+                description += ('last category are detections that do not overlap with labeled data'
+                                ' and labels that have not been detected. ')
+
+        description += f'Showing {self.categories_to_display} of {self.num_classes} classes:'
+
+        fig = px.imshow(display_confusion_matrix,
+                        x=categories,
+                        y=categories,
+                        text_auto=True)
+
+        fig.update_layout(width=600, height=600)
+        fig.update_xaxes(title='Predicted Value', type='category')
+        fig.update_yaxes(title='True value', type='category')
+
+        return CheckResult(
+            self.matrix,
+            header='Confusion Matrix',
+            display=[description, fig]
+        )
+
+    def update_object_detection(self, predictions, labels):
+        for image_detections, image_labels in zip(predictions, labels):
+            try:
+                detections_passed_threshold = [
+                    detection for detection in image_detections if detection[4] > self.confidence_threshold
+                ]
+            except IndexError:
+                # detections are empty, update matrix for labels
+                for label in image_labels:
+                    gt_class = label[0]
+                    self.matrix[self.num_classes, gt_class] += 1
+                continue
```

Comment:
Not a programmer, but isn't try-except a relatively costly process? Wouldn't it be better to just check if detection[4] exists? also, doesn't this mean that you don't continue with creating the relevant list, thus skipping the detections after the empty one?